azure-pipelines.yml

```diff
@@ -0,0 +1,209 @@
+    displayName: publish self-contained apiport - win7-x64
+
+  - script: dotnet publish src/ApiPort/ApiPort/ApiPort.csproj -r win7-x86 -f netcoreapp2.1 --self-contained true
+    displayName: publish self-contained apiport - win7-x86
+
+  - powershell: |
+      # Extract rule Ids into a plaintext file that is expected by the Roslyn Analyzer build task
+      $xdoc = [xml](Get-Content ".\rules.ruleset")
+      $xdoc.RuleSet.Rules.Rule.Id | Out-File "bin\RoslynAnalyzerSuppressions.txt"
+
+      # Add NoWarn Ids
+      $xdoc = [xml](Get-Content ".\Directory.Build.props")
+      $xdoc.Project.PropertyGroup.NoWarn -split "," | Select -skip 1 | Where {$_} | Add-Content ".\bin\RoslynAnalyzerSuppressions.txt"
+    displayName: Extract suppression ids for Roslyn Analyzer task
+
+  - task: VSTest@2
+    displayName: Test with VSTest
+    inputs:
+      testAssemblyVer2: |
+        **\*test*.dll
+        !**\obj\**
+        !**\PortabilityServiceIntegrationTests.dll
+      runOnlyImpactedTests: false
+      vsTestVersion: '16.0'
+      runInParallel: false
+      runTestsInIsolation: false
+      codeCoverageEnabled: false
+      platform: $(BuildPlatform)
+      configuration: $(BuildConfiguration)
+      diagnosticsEnabled: True
+
+  - task: DotNetCoreCLI@2
+    displayName: Test with dotnet test
+    inputs:
+      command: test
+      projects: |
+        tests/**/*[Tt]ests/*.csproj
+        !tests/ApiPort/ApiPortVS.Tests/ApiPortVS.Tests.csproj
+        !**\PortabilityServiceIntegrationTests.dll
+      arguments: --configuration $(BuildConfiguration) --no-build
+
+  - task: CopyFiles@2
+    displayName: Copy Analysis files to Staging
+    inputs:
+      SourceFolder: bin\$(BuildConfiguration)
+      Contents: |
+        ApiPort\**\*
+        ApiPort.Offline\**\*
+        ApiPort.Vsix\*.vsix
+        raw\**\*
+        *.binlog
+      TargetFolder: $(Build.StagingDirectory)\drop
+
+  - task: CopyFiles@2
+    displayName: Collection symbol files
+    inputs:
+      SourceFolder: bin\$(BuildConfiguration)
+      Contents: '**\*.pdb'
+      TargetFolder: $(Build.StagingDirectory)\symbols
+
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-roslynanalyzers.RoslynAnalyzers@2
+    displayName: Run Roslyn Analyzers
+    inputs:
+      suppressionFileForCompilerWarnings: bin\RoslynAnalyzerSuppressions.txt
+
+  - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
+    displayName: Component Detection
+
+  - publish: $(Build.StagingDirectory)\drop
+    displayName: Publish drop
+    artifact: drop
+    enabled: false
+
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-fxcop.FxCop@2
+    displayName: 'Run FxCop'
+    inputs:
+      inputType: Basic
+      targets: 'bin\$(BuildConfiguration)\ApiPort*\net4*\**\ApiPort*.*;bin\$(BuildConfiguration)\ApiPort.Offline\net4*\**\Microsoft.Fx.*.dll;'
```

Comment:
I didn't make this up. This came directly from your existing pipeline, so I can't say why it was this way.
That said, in my experience FxCop fails miserably on anything other than .NET Framework targeted assemblies. So I can imagine why it was done this way. 
Further: I recommend we drop FxCop entirely since it's a deprecated thing. We should switch to the analyzer alternative (in a subsequent PR).